packages/common/src/server.ts

```diff
@@ -358,28 +358,43 @@ export class YogaServer<
 
     const headers: Record<string, string> = {}
 
-    const currentOrigin = request.headers.get('origin')
-
-    headers['Access-Control-Allow-Origin'] = '*'
-
-    if (currentOrigin) {
-      const credentialsAsked = request.headers.get('cookies')
-      if (credentialsAsked || corsOptions.credentials !== false) {
-        headers['Access-Control-Allow-Origin'] = currentOrigin
-      }
-    }
-
+    // If defined origins have '*' or undefined by any means, we should allow all origins
     if (
-      currentOrigin != null &&
-      corsOptions.origin?.length &&
-      !corsOptions.origin.includes(currentOrigin) &&
-      !corsOptions.origin.includes('*')
+      corsOptions.origin == null ||
```

Comment:
```suggestion
      corsOptions.origin === null ||
```

Reply:
`foo == null`  covers null and undefined - was this the intension here?

Reply:
Yes 👍